src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java

```diff
@@ -593,6 +605,53 @@ public Collection<Flow> findLoopedByLoopSwitchId(SwitchId switchId) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public Map<Integer, List<Flow>> findSwitchFlowsByPort(SwitchId switchId, Collection<Integer> ports) {
+        try {
+            @SuppressWarnings("unchecked")
+            Map<Integer, List<String>> portToFlowIdMap = collectToPortToFlowMap(framedGraph().traverse(g -> g.V()
+                            .hasLabel(PathSegmentFrame.FRAME_LABEL)
+                            .or(
+                                    has(PathSegmentFrame.SRC_SWITCH_ID_PROPERTY, switchId.toString()),
+                                    has(PathSegmentFrame.DST_SWITCH_ID_PROPERTY, switchId.toString()))
+                            .as(PATH_SEGMENT_ALIAS)
+                            .in("owns")
+                            .in("owns")
+                            .hasLabel(FlowFrame.FRAME_LABEL)
+                            .as(FLOWS_ALIAS)
+                            .select(PATH_SEGMENT_ALIAS, FLOWS_ALIAS)
+                    ).getRawTraversal().fill(new ArrayList<>()).stream().map(x -> (Map<String, Vertex>) x),
+                    PathSegmentFrame.SRC_PORT_PROPERTY, PathSegmentFrame.DST_PORT_PROPERTY);
+
+            Map<String, Flow> flowIdToFlowMap = findFlowsByFlowIds(portToFlowIdMap.values()
+                    .stream()
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet()))
+                    .stream()
+                    .collect(Collectors.toMap(Flow::getFlowId, flow -> flow));
+
+            Predicate<Map.Entry<Integer, List<String>>> portsPredicate =
+                    (ports != null && ports.size() > 0) ? e -> ports.contains(e.getKey()) : e -> true;
+
+            return portToFlowIdMap.entrySet().stream()
+                    .filter(portsPredicate)
+                    .collect(Collectors.toMap(Map.Entry::getKey,
+                            entry -> entry.getValue().stream()
+                                    .distinct()
+                                    .map(flowIdToFlowMap::get)
+                                    .collect(Collectors.toList()),
+                            (l1, l2) ->
+                                    Stream.of(l1, l2).flatMap(Collection::stream)
+                                            .sorted()
+                                            .distinct()
+                                            .collect(Collectors.toList()),
+                                    TreeMap::new));
+        } catch (ClassCastException | NumberFormatException e) {
+            log.debug("An exception in findSwitchFlowsByPort: {}", e.getMessage());
```

Comment:
Why is it debug?

Reply:
My reasoning is that an exception here is something not normal/recoverable; this is not going to be happening on daily basis. The scenario of usage is when something went wrong, but the error appears like "time out" error, then we want to debug to find out more details, or in opposite way: to make sure this exception is not thrown.
Examples why the code ends up here are: ferma frramework returns an object of different type or a port is not a number and cannot be parsed. I believe both are unlikely to happen.
Why not an error type? It could be, however, the exception is re-thrown, so the error is reported in a standard way. It might be tricky to find though, so I added the key word in the message to make it easier.

Reply:
I,d bump the log level to error/exception tbh. The problem here is the following: this code is very core and it's not guaranteed every place it will be called will have proper logging/error handling. The exception should be rare, which makes things even more complicated. If it fire one in prod, it might be very tricky to reproduce, and not having that logs will be very painful